src/diffusers/models/attention_processor.py

```diff
@@ -291,6 +311,29 @@ def prepare_attention_mask(self, attention_mask, target_length, batch_size=None)
             attention_mask = attention_mask.repeat_interleave(head_size, dim=0)
         return attention_mask
 
+    def prepare_encoder_hidden_states(self, hidden_states, encoder_hidden_states=None):
+        if encoder_hidden_states is None:
+            return hidden_states
+
+        if self.norm_cross is None:
+            return encoder_hidden_states
+
+        if isinstance(self.norm_cross, nn.LayerNorm):
+            encoder_hidden_states = self.norm_cross(encoder_hidden_states)
+        elif isinstance(self.norm_cross, nn.GroupNorm):
+            # Group norm norms along the channels dimension and expects
+            # input to be in the shape of (N, C, *). In this case, we want
+            # to norm along the hidden dimension, so we need to move
+            # (batch_size, sequence_length, hidden_size) ->
+            # (batch_size, hidden_size, sequence_length)
+            encoder_hidden_states = encoder_hidden_states.transpose(1, 2)
+            encoder_hidden_states = self.norm_cross(encoder_hidden_states)
+            encoder_hidden_states = encoder_hidden_states.transpose(1, 2)
+        else:
+            assert False
+
+        return encoder_hidden_states
+
```

Comment:
Standardized encoder hidden states preprocessing

Reply:
Was this method renamed to `norm_encoder_hidden_states()`?

Reply:
Nevermind. 

https://github.com/huggingface/diffusers/pull/3021/commits/9160e517b3bf8ee0b6ca4923c49c928f2a657ef9

Reply:
yessir!